### PR TITLE
Update https-proxy-agent for CVE

### PR DESCRIPTION
**Pull request by apassant**
#### Why?

Fixes https://github.com/intercom/intercom/issues/141597

### CODE REVIEW

**Comment by SeanHealy33** [comment]
I'm confused where is proxy agent in the yarn lock?

**Comment by apassant** [comment]
```https-proxy-agent@^2.2.0:
  version "2.2.1"
  resolved "https://registry.yarnpkg.com/https-proxy-agent/-/https-proxy-agent-2.2.1.tgz#51552970fa04d723e04c56d04178c3f92592bbc0"
  integrity sha512-HPCTS1LW51bcyMYbxUIOO4HEOlQ1/1qRaFWcyxvwaqUS9TY88aoEuHUY33kuAh1YhVVaDQhLZsnPd+XNARWZlQ==
  dependencies:
    agent-base "^4.1.0"
    debug "^3.1.0"```

File: package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "bluebird": "^3.3.4",
     "htmlencode": "^0.0.4",
     "lodash": "^4.17.11",
-    "request": "^2.83.0"
+    "request": "^2.83.0",
+    "https-proxy-agent": "^2.2.0"
```
Comment (SeanHealy33) [inline package.json]: I think we can remove this now?

File: package.json [hunk 2]
```diff
@@ -39,6 +40,9 @@
     "nock": "7.5.0",
     "sinon": "^6.1.4"
   },
+  "resolutions": {
```
Comment (SeanHealy33) [inline package.json]: I wasnt aware this was a thing npm could do. Neat <3

File: yarn.lock [hunk 3]
```diff
@@ -2049,14 +2067,13 @@ http-signature@~1.2.0:
     jsprim "^1.2.2"
     sshpk "^1.7.0"
 
-https-proxy-agent@^1.0.0:
-  version "1.0.0"
-  resolved "https://registry.yarnpkg.com/https-proxy-agent/-/https-proxy-agent-1.0.0.tgz#35f7da6c48ce4ddbfa264891ac593ee5ff8671e6"
-  integrity sha1-NffabEjOTdv6JkiRrFk+5f+GceY=
+https-proxy-agent@^1.0.0, https-proxy-agent@^2.2.0:
```
Comment (mattdonnelly) [inline yarn.lock]: I think Sean is right, in order to fix this we need to upgrade `gulp-nsp` to a version which doesnt use the affected  package. This line indicates we have both 1.0.0 _and_ 2.2.0 now

Comment (mattdonnelly) [reply yarn.lock]: Looks like NSP has been retired https://github.com/nodesecurity/nsp - I think we should just remove the package and delete its usage from the gulpfile

Comment (mattdonnelly) [reply yarn.lock]: Also we don't really need it anymore given Github now does this for us

File: yarn.lock [hunk 4]
```diff
@@ -43,29 +51,49 @@ acorn-to-esprima@^2.0.4:
...
+    fast-json-stable-stringify "^2.0.0"
+    json-schema-traverse "^0.4.1"
+    uri-js "^4.2.2"
 
 amdefine@>=0.0.4:
   version "1.0.1"
   resolved "https://registry.yarnpkg.com/amdefine/-/amdefine-1.0.1.tgz#4a5282ac164729e93619bcfd3ad151f817ce91f5"
   integrity sha1-SlKCrBZHKek2Gbz9OtFR+BfOkfU=
 
+ansi-colors@^1.0.1:
```
Comment (SeanHealy33) [inline yarn.lock]: actually it does seem odd we are removing a module and getting additions is this correct?

File: yarn.lock [hunk 5]
```diff
@@ -2,31 +2,39 @@
 # yarn lockfile v1
 
 
-"@sinonjs/commons@^1.0.2":
-  version "1.3.0"
-  resolved "https://registry.yarnpkg.com/@sinonjs/commons/-/commons-1.3.0.tgz#50a2754016b6f30a994ceda6d9a0a8c36adda849"
-  integrity sha512-j4ZwhaHmwsCb4DlDOIWnI5YyKDNMoNThsmwEpfHx6a1EpsGZ9qYLxP++LMlmBRjtGptGHFsGItJ768snllFWpA==
+"@sinonjs/commons@^1", "@sinonjs/commons@^1.0.2":
```
Comment (mattdonnelly) [inline yarn.lock]: We shouldn't see any new packages added after removing `gulp-nsp`. Might need to revert the yarn.lock changes and then retry removing the package

**Comment by apassant** [comment]
https://github.com/intercom/intercom-node/pull/239

**Pull request closed without merging**